src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpProxyCacheableRequestGroup.java

```diff
@@ -100,6 +100,21 @@ else if (!groupRequest.canDeferRequest(request))
             final String newIfNoneMatch = groupRequest.withIfNoneMatch(request.ifNoneMatch);
             doRequest(request, newIfNoneMatch);
         }
+        else if (!attachedResponses.isEmpty())
+        {
+            final String etag = cacheEntry.etag();
+            final boolean notModified = etag != null && etag.equals(request.ifNoneMatch);
+            final long traceId = factory.supplyTraceId.getAsLong();
+            if (notModified)
+            {
+                request.doNotModifiedResponse(traceId);
+            }
+            else
+            {
+                request.doCachedResponse(Instant.now(), traceId);
+            }
+            queuedRequests.clear();
```

Comment:
Should this be `queuedRequests.remove(request)` instead?

Reply:
yep